### PR TITLE
Add new section in Qibolab website

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
With this PR, I am taking into account the issue #186.
-  It includes CI tests for linux, mac and windows and for multiple version of python. 
-  A section **How to connect Qibolab to your lab** is included in the documentation, where you can find an easy example. 
- I also added the extension `sphinx_copybutton`, in this way code blocks have little copy buttons.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: doc/source/developer-guides/general.rst [hunk 1]
```diff
@@ -11,8 +11,13 @@ New platforms can be implemented by inheriting the
 :class:`qibolab.abstract.AbstractPlatform` and implementing its abstract
 methods. In particular the developer should:
 
-* Perform an ``AbstractPlatform`` inheritance.
-* Create a yaml in ``src/qibolab/runcards/`` containing all the instruments and the calibration parameters
+* Perform an ``AbstractPlatform`` inheritance, we suggest saving it in ``src/qibolab/platforms/``, if you want to include it in the ``Qibolab`` library.
```
Comment (andrea-pasquale) [inline doc/source/developer-guides/general.rst:14]: To be consistent with what you have on the second point you could specify that your child class is `ExamplePlatform` and that you are coding it in the file `src/qibocal/platforms/example.py`.

File: doc/source/developer-guides/general.rst [hunk 2]
```diff
@@ -21,24 +26,35 @@ methods. In particular the developer should:
...
+.. code-block:: yaml
+
     settings:
         delay_before_readout: 0
         delay_between_pulses: 0
         resonator_spectroscopy_max_ro_voltage: 5726.62338856
         rabi_oscillations_pi_pulse_min_voltage: 5291.34802850
         experiment_start_instrument: awg
 
+All the information useful to connect and set up the instruments are collected
```
Comment (andrea-pasquale) [inline doc/source/developer-guides/general.rst:53]: I think that `settings` contains also general information about the platform including the topology and not only information related to the instruments. I would suggest to modify this sentence.

File: doc/source/getting-started/lab.rst [hunk 3]
```diff
@@ -0,0 +1,164 @@
+How to connect Qibolab to your lab
+==================================
+
+In this section we will see how to let ``Qibolab`` communicate with your lab’s instruments and run an experiment.
```
Comment (andrea-pasquale) [inline doc/source/getting-started/lab.rst:4]:
```suggestion
In this section we will show how to let ``Qibolab`` communicate with your lab’s instruments and run an experiment.
```

File: doc/source/getting-started/lab.rst [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                resonator_freq: 7824855425
+                qubit_freq: 5082441303
+                T1: 21160
+                T2: 17214
+                state0_voltage: 847
+                state1_voltage: 251
+                mean_gnd_states: (-0.0008313786783245074+0.00044815319987281076j)
+                mean_exc_states: (-0.00022773024569023835+0.0003203066357136082j)
+
+The class ``DummyPlatform`` has a method ``execute_pulse_sequence`` that reads the
```
Comment (andrea-pasquale) [inline doc/source/getting-started/lab.rst:82]:
```suggestion
The class ``DummyPlatform`` has a method ``execute_pulse_sequence`` that executes the
```

File: doc/source/getting-started/lab.rst [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+input pulse sequences and return an output pulse sequence.
```
Comment (andrea-pasquale) [inline doc/source/getting-started/lab.rst:83]:
```suggestion
input pulse sequences and return the readout results.
```

File: doc/source/getting-started/lab.rst [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+.. code-block:: python
+
+    class DummyPlatform(AbstractPlatform):
+        #Dummy platform that returns random voltage values.
+        #Args:
+        #    name (str): name of the platform.
```
Comment (andrea-pasquale) [inline doc/source/getting-started/lab.rst:90]: We are missing a comment line for the runcard, correct?

**Review by andrea-pasquale — COMMENTED** [review]
Thanks for implementing this @Edoardo-Pedicillo.
We are missing the `furo` dep in the `setup.py`, correct? I think that we forgot to add it in #210.

**Comment by Edoardo-Pedicillo** [comment]
Thanks, @andrea-pasquale for your review. I agree about `furo` and I have updated the `conf.py`. 
To be 100% sure I have done the right thing, I ask @MatteoRobbiati to check it.

**Review by MatteoRobbiati — COMMENTED** [review]
Thank you @Edoardo-Pedicillo for this. 
The `furo` dep is correctly placed. 

I also like the schema you draw, maybe I would prefer thicker arrows, but these are purely personal preferences.

Very nice page.

**Comment by Edoardo-Pedicillo** [comment]
Thanks @MatteoRobbiati. I have changed the arrows.

**Review by MatteoRobbiati — COMMENTED** [review]
Thank you @Edoardo-Pedicillo !!

**Comment by scarrazza** [comment]
Shall we merge this?

**Comment by andrea-pasquale** [comment]
> Shall we merge this?
I think so. Probably this should point to `main`, right @Edoardo-Pedicillo?

**Review by andrea-pasquale — APPROVED** [review]

**Comment by Edoardo-Pedicillo** [comment]
> Shall we merge this?

Yes, it is ready.

> I think so. Probably this should point to main, right @Edoardo-Pedicillo?

Yes. Thank @andrea-pasquale  and @MatteoRobbiati for the review.

**Pull request merged**